climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
+    for i in range(1, len(x)):
+        x[i] = _h_ac(x[i - 1], x[i], theta)
+    # scale x to have magnitude [0,1]
+    x = (x + np.sqrt(3)) / (2 * np.sqrt(3))
+    # resample at target time step
+    x_ts = np.interp(np.arange(start=0, stop=n_ts_hourly_exact, step=time_step_h),
+                     np.arange(n_ts_hourly), x)
+    return x_ts
+
+
+@jit
+def _h_ac(x, y, theta):
+    # https://stats.stackexchange.com/questions/48086/
+    #   algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    # Note that the definition of Gamma is not very clear there, but the
+    # formulation below does what the text says.
+    # Tests indicated that this works, while the following fails:
+    # gamma = np.abs(np.mod(np.mod(theta, np.pi), np.pi / 2) - np.pi / 4)
+    gamma = np.abs(np.mod(theta, np.pi) - \
+                   np.floor((np.mod(theta, np.pi) / (np.pi / 2)) + 0.5) * np.pi / 2)
+    # np.abs(dat['theta2'] - np.floor((dat['theta2'] / (np.pi / 2)) + 0.5) * np.pi / 2)
+    return 2 * np.sqrt(3) * (_f_ac(np.cos(theta) * x + np.sin(theta) * y, gamma) - 1 / 2)
+
+
+@jit
+def _f_ac(z, theta):
+    """F transform for autocorrelated random uniform series generation
+
+    This is function F defined here:
+    https://stats.stackexchange.com/questions/48086/algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    The CDF of Y.
+
+    Parameters:
+        z (float): longitude coordinates, in decimal degrees
+        theta (float): arccos of the autocorrelation
+
+    Returns:
+        float (CDF at z)
+    """
+    c = np.cos(theta)
+    s = np.sin(theta)
+    if z >= np.sqrt(3) * (c + s):
+        res = 1
+    elif z > np.sqrt(3) * (c - s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (-3 - z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 9 * np.sin(2 * theta))
+    elif z > np.sqrt(3) * (-c + s):
+        res = 1 / 6 * (3 + np.sqrt(3) * z / c)
+    elif z > -np.sqrt(3) * (c + s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 3 * (1 + np.sin(2 * theta)))
+    else:
+        res = 0
+    return res
+
+
+@jit
+def _get_bearing_angle(lon, lat):
+    """Compute bearing angle of great circle paths defined by consecutive points
+
+    Returns initial bearing (also called forward azimuth) of the n-1 great circle
+    paths define by n consecutive longitude/latitude points. The bearing is the angle
+    (clockwise from North) which if followed in a straight line along a great-circle
+    arc will take you from the start point to the end point. See also:
+    http://www.movable-type.co.uk/scripts/latlong.html
+    Here, the bearing of each pair of consecutive points is computed.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        bearing_angle (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
```

Comment:
According to CLIMADA's coding conventions, docstrings should follow the NumPy style. From one of the developer meetings, I remember that, while older code in CLIMADA is still allowed to contain google-style docstrings, new code is required to follow the new convention. Old docstrings should be converted to the new convention whenever they are touched.

Reply:
yes e837bc88958cd756867fef5a7ef8d79571c8981c